crates/dt-workspace/src/config/mod.rs

```diff
@@ -0,0 +1,80 @@
+use std::path::PathBuf;
+
+#[cfg(feature = "cli")]
+use cli_config::CliConfig;
+use env_config::EnvConfig;
+use toml_config::TomlConfig;
+
+#[cfg(feature = "cli")]
+pub mod cli_config;
+pub mod env_config;
+pub mod toml_config;
+
+macro_rules! config_field {
+  ($start:ident, $($fallback:ident),+; $field:ident) => {
+      $start.and_then(|value| value.$field)
+        $(
+        .or_else(|| config_field!($fallback; $field))
+        )+
+  };
+  ($struct:ident; $field:ident) => {
+    $struct.and_then(|value| value.$field)
+  }
+}
+
+#[derive(Debug, Default, PartialEq)]
+pub struct CombinedConfig {
+    include_paths: Vec<PathBuf>,
+}
+
+impl CombinedConfig {
+    #[must_use]
+    pub fn merge(
+        toml: Option<TomlConfig>,
+        #[cfg(feature = "cli")] cli: Option<CliConfig>,
+        env: Option<EnvConfig>,
+    ) -> Self {
+        Self {
+            #[cfg(feature = "cli")]
+            include_paths: config_field!(cli, env, toml; include_paths).unwrap_or_default(),
+            #[cfg(not(feature = "cli"))]
+            include_paths: config_field!(env, toml; include_paths).unwrap_or_default(),
```

Comment:
Use the new macro as such:
```suggestion
            include_paths: config_field!(cli, env, toml; include_paths).unwrap_or_default(),
```